### PR TITLE
Format link after hitting newline

### PR DESCRIPTION
**Pull request by zaynetro**
Works in the block and outside the block.

[Screencast from 2023-01-06 21-10-53.webm](https://user-images.githubusercontent.com/627197/211082729-4ee70c20-d2c7-46ab-a2ae-20babe966cfe.webm)

### CODE REVIEW

**Comment by cgestes** [comment]
I am always careful when there are changes involving delta...

had a quick look, it seems the logic is fine. nice :)

**Comment by zaynetro** [comment]
How can we move this forward?

**Comment by amantoux** [comment]
I'll try to have a look this week end

File: packages/parchment/lib/src/heuristics/insert_rules.dart [hunk 1]
```diff
@@ -137,16 +137,18 @@ class ResetLineFormatOnNewLineRule extends InsertRule {
...
       if (target.attributes != null &&
           target.attributes!.containsKey(ParchmentAttribute.heading.key)) {
-        resetStyle = ParchmentAttribute.heading.unset.toJson();
+        final resetStyle = ParchmentAttribute.heading.unset.toJson();
+        return Delta()
+          ..retain(index)
+          ..insert('\n', target.attributes)
+          ..retain(1, resetStyle)
+          ..trim();
+      } else {
```
Comment (amantoux) [inline packages/parchment/lib/src/heuristics/insert_rules.dart:148]: Unnecessary?

Comment (zaynetro) [reply packages/parchment/lib/src/heuristics/insert_rules.dart]: Indeed! Removed

**Review by amantoux — CHANGES_REQUESTED** [review]
Many thanks for this PR
On small change request

**Review by amantoux — APPROVED** [review]
LGTM
Many thanks!

**Pull request merged**